src/components/IntlTelInput.d.ts

```diff
@@ -0,0 +1,494 @@
+    fullNumber: string,
+    isValid: boolean,
+  ) => void
+  /**
+   * Disable this component.
+   * @default false
+   */
+  disabled?: boolean
+  /**
+   * Static placeholder for input controller. When defined it takes priority over autoPlaceholder.
+   */
+  placeholder?: string
+  /**
+   * Enable auto focus
+   * @default false
+   */
+  autoFocus?: boolean
+  /**
+   * Set the value of the autoComplete attribute on the input.
+   * For example, set it to phone to tell the browser where to auto complete phone numbers.
+   * @default 'off'
+   */
+  autoComplete?: string
+  /**
+   * Style object for the wrapper div. Useful for setting 100% width on the wrapper, etc.
+   */
+  style?: React.CSSProperties
+  /**
+   * Render fullscreen flag dropdown when mobile useragent is detected.
+   * The dropdown element is rendered as a direct child of document.body
+   * @default true
+   */
+  useMobileFullscreenDropdown?: boolean
+  /**
+   * Pass through arbitrary props to the tel input element.
+   * @default {}
+   */
+  telInputProps?: React.InputHTMLAttributes<HTMLInputElement>
+  /**
+   * Format the number.
+   * @default true
+   */
+  format?: boolean
+  /**
+   * Allow main app to do things when flag icon is clicked.
+   * @default null
+   */
+  onFlagClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void
+}
+
+export interface IntlTelInputState {
+  showDropdown: boolean
+  highlightedCountry: number
+  value: string
+  disabled: boolean
+  readonly: boolean
+  offsetTop: number
+  outerHeight: number
+  placeholder: string
+  title: string
+  countryCode: string
+  dialCode: string
+  cursorPosition: any
+}
+
+export default class IntlTelInput extends React.Component<
+  IntlTelInputProps,
+  IntlTelInputState
+> {
+  //#region Properties
+  wrapperClass: {
+    [key: string]: boolean
+  }
+
+  defaultCountry?: string
+
+  autoCountry: string
+
+  tempCountry: string
+
+  startedLoadingAutoCountry: boolean
+
+  dropdownContainer?: React.ElementType | ''
+
+  isOpening: boolean
+
+  isMobile: boolean
+
+  preferredCountries: CountryData[]
+
+  countries: CountryData[]
+
+  countryCodes: {
+    [key: string]: string[]
+  }
+
+  windowLoaded: boolean
+
+  query: string
+
+  selectedCountryData?: CountryData
+
+  // prop copies
+  autoHideDialCode: boolean
+
+  nationalMode: boolean
+
+  allowDropdown: boolean
+
+  // refs
+  flagDropDown: HTMLDivElement | null
+
+  tel: HTMLInputElement | null
+
+  // NOTE:
+  // The underscore.deferred package doesn't have known type definitions.
+  // The closest counterpart is jquery's Deferred object, which it claims to derive itself from.
+  // These two are equivalent if you log it in console:
+  //
+  // underscore.deferred
+  // var deferred = new _.Deferred()
+  //
+  // jquery
+  // var deferred = $.Deferred()
+  deferreds: JQuery.Deferred<any, any, any>[]
+
+  autoCountryDeferred: JQuery.Deferred<any, any, any>
+
+  utilsScriptDeferred: JQuery.Deferred<any, any, any>
```

Comment:
I've considered writing the `underscore.deferred` definitions by hand so that `@types/jquery` won't be needed. However, the definition of `JQuery.Deferred` is more massive than it's really worth to copy over, so I chose to simply reuse it.